src/components/ClusterOverview/TopConsumers/TopConsumers.js

```diff
@@ -1,100 +1,188 @@
+          case BY_STORAGE:
+            return getMetric(this.props.workloadStorageResults, 'pod_name', formatBytes);
+          case BY_NETWORK:
+            return getMetric(this.props.workloadNetworkResults, 'pod_name');
+          default:
+            // eslint-disable-next-line no-console
+            console.log(`Unknown metric ${this.state.sortBy.name}`);
+            return null;
+        }
+      case NODES:
+        switch (this.state.sortBy.name) {
+          case BY_CPU:
+            return getMetric(this.props.infraCpuResults, 'node', cpu => formatCpu(cpu, 1));
+          case BY_MEMORY:
+            return getMetric(this.props.infraMemoryResults, 'node', formatBytes);
+          case BY_STORAGE:
+            return getMetric(this.props.infraStorageResults, 'node', formatBytes);
+          case BY_NETWORK:
+            return getMetric(this.props.infraNetworkResults, 'node');
+          default:
+            // eslint-disable-next-line no-console
+            console.log(`Unknown metric ${this.state.sortBy.name}`);
+            return null;
+        }
+      default:
+        // eslint-disable-next-line no-console
+        console.log(`Unknown metric type ${this.state.type.name}`);
+        return null;
+    }
+  };
 
   render() {
+    const typeDropdown = {
+      id: 'type-dropdown',
+      type: DROPDOWN,
+      choices: metricTypes,
+      onChange: newValue => this.setState({ type: newValue }),
+      value: this.state.type,
+      className: 'kubevirt-consumers__dropdown-filter',
+    };
     const sortByDropdown = {
       id: 'sort-by-dropdown',
       type: DROPDOWN,
-      choices: ['CPU', 'Memory', 'Network', 'Storage'],
-      onChange: newValue => this.onDropdownChange('sortBy', newValue),
+      choices: sortBy,
+      onChange: newValue => this.setState({ sortBy: newValue }),
       value: this.state.sortBy,
+      className: 'kubevirt-consumers__dropdown-filter',
     };
-    const filterDropdown = {
-      id: 'filter-dropdown',
-      type: DROPDOWN,
-      choices: ['All', PodModel.kind, VirtualMachineModel.kind],
-      onChange: newValue => this.onDropdownChange('filter', newValue),
-      value: this.state.filter,
-    };
+
+    const metric = this.getCurrentMetric();
+
     return (
-      <div>
-        <div className="kubevirt-consumers__filters">
-          {getFormElement(filterDropdown)}
-          {getFormElement(sortByDropdown)}
-        </div>
-        {sortConsumers(this.props.metrics, this.state.filter, this.state.sortBy).map((c, index) => (
-          <UtilizationBar key={index} {...c} className="kubevirt-consumers__bar" />
-        ))}
-      </div>
+      <DashboardCard>
+        <DashboardCardHeader>
+          <DashboardCardTitle>Top Consumers</DashboardCardTitle>
+          <DashboardCardTitleHelp>help for top consumers</DashboardCardTitleHelp>
+        </DashboardCardHeader>
+        <DashboardCardBody>
+          <ConsumersFilter>
+            {getFormElement(typeDropdown)}
```

Comment:
would be maybe good to add some margin between them

Reply:
added

![space](https://user-images.githubusercontent.com/2078045/55627539-b854c500-57ae-11e9-8990-0ccbc710fa4d.png)